urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java

```diff
@@ -836,4 +850,51 @@ static void checkOptIn(Context context) {
         }
     }
 
+    /**
+     * Helper method to parse a String array into {@link PrivacyManager.Feature} array.
+     * @param features The String features to parse.
+     * @return The {@link PrivacyManager.Feature} array.
+     */
+    @PrivacyManager.Feature
+    private int[] parseStringFeatures(ReadableArray features) {
+        @PrivacyManager.Feature
+        int[] intFeatures = new int[features.size()];
+
+        for (int i = 0; i < features.size(); i++) {
+            switch (features.getString(i)) {
+                case "FEATURE_NONE":
+                    intFeatures[i] = PrivacyManager.FEATURE_NONE;
+                    break;
+                case "FEATURE_IN_APP_AUTOMATION":
+                    intFeatures[i] = PrivacyManager.FEATURE_IN_APP_AUTOMATION;
+                    break;
+                case "FEATURE_MESSAGE_CENTER":
+                    intFeatures[i] = PrivacyManager.FEATURE_MESSAGE_CENTER;
+                    break;
+                case "FEATURE_PUSH":
+                    intFeatures[i] = PrivacyManager.FEATURE_PUSH;
+                    break;
+                case "FEATURE_CHAT":
+                    intFeatures[i] = PrivacyManager.FEATURE_CHAT;
+                    break;
+                case "FEATURE_ANALYTICS":
+                    intFeatures[i] = PrivacyManager.FEATURE_ANALYTICS;
+                    break;
+                case "FEATURE_TAGS_AND_ATTRIBUTES":
+                    intFeatures[i] = PrivacyManager.FEATURE_TAGS_AND_ATTRIBUTES;
+                    break;
+                case "FEATURE_CONTACTS":
+                    intFeatures[i] = PrivacyManager.FEATURE_CONTACTS;
+                    break;
+                case "FEATURE_LOCATION":
+                    intFeatures[i] = PrivacyManager.FEATURE_LOCATION;
+                    break;
+                default:
+                    intFeatures[i] = PrivacyManager.FEATURE_ALL;
```

Comment:
I think it would be safer to add a case for "FEATURE_ALL", and make the default label log an error. otherwise if someone is using plain javascript and they pass a string that's not in this list it would be interpreted as FEATURE_ALL, which is probably not what we want.

Reply:
Good catch, didn't think about it. Thank you very much 👍